src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -425,162 +426,95 @@ internal void CheckConstraintLanguageVersionAndRuntimeSupportForConversion(Synta
             bool wasCompilerGenerated,
```

Comment:
Can you remind me, when is a collection literal expression compiler-generated?

Reply:
`wasCompilerGenerated` refers to the conversion rather than the collection literal, and the conversion is considered compiler generated in cases such as nested elements for instance: `int[][] a = [[1], [2, 3]];`.

Reply:
That seems weird, but I can accept it :)